Sources/NextcloudFileProviderKit/Item/Item+Modify.swift

```diff
@@ -836,6 +836,16 @@ public extension Item {
             return (nil, NSFileProviderError(.noSuchItem))
         }
 
+        guard metadata.classFile != "lock", !isLockFileName(metadata.fileName) else {
+            Self.logger.info(
+                """
+                System requested modification of lock file \(self.filename, privacy: .public)
+                    Marking as complete without syncing to server.
+                """
+            )
+            return (self, nil)
```

Comment:
This will lead to bad metadata on the file; a new item should be returned with the changes provided by the system to all relevant fields